File: java/CreditCardValidator.java
import java.util.Arrays;
import java.util.ArrayList;
import java.util.Scanner;
public class CreditCardValidator{
public static void main (String[] args){
Scanner scanner = new Scanner(System.in);
mainMenu();

}

public static void mainMenu(){
Scanner scanner = new Scanner (System.in);
System.out.println("Hello, Kindly Enter Card details to verify");
String input = scanner.nextLine();
String [] split = splitIntoArrays(input);
System.out.printf("**Credit Card Type: %s%n",checkCardType(split));
System.out.printf("**Credit Card Number: %s%n", input);
System.out.printf("**Credit Digit Length: %d%n", split.length);
System.out.printf("**Credit Card Validity Status: %s%n", checkCardValidityStatus(split));
System.out.printf("**************************************");



}
public static String[] splitIntoArrays(String numbers){

String [] cardNumbers = numbers.split("");
return cardNumbers;

}

public static String checkCardValidityStatus(String[] numbers){

if (numbers.length < 13 || numbers.length > 16) return "Invalid";

int [] newArray = new int [16];
ArrayList<Integer> AddArray = new ArrayList<Integer>();
for (int count = numbers.length -1 ; count >= 0; count--){
if (count % 2 == 0){

newArray[count] = Integer.parseInt(numbers[count]);
int doubleNumber = newArray[count] * 2;

if (doubleNumber >= 10){
int firstNum = doubleNumber % 10;
int divideNum = doubleNumber / 10;
int secondNum = divideNum % 10;
int sum = firstNum + secondNum ;
AddArray.add(sum);
}else{
AddArray.add(doubleNumber); 
}

}



}
int result = 0;
for (int counter = 0; counter < AddArray.size(); counter++){
result += AddArray.get(counter); 

}

int [] oddPlaces = new int [16];
int sumOdd = 0;
for (int countOddPlaces = numbers.length - 1; countOddPlaces >= 0; countOddPlaces--){
if (countOddPlaces % 2 != 0){
sumOdd += Integer.parseInt(numbers[countOddPlaces]);


}


} 
int totalSum = sumOdd + result;

if (totalSum % 10 == 0) return "Valid";
return "Invalid";

}




public static String checkCardType(String[] numbers){

if (numbers[0].equals("5")) return "Master Card";
else
if (numbers[0].equals("4")) return "Visa Card";
else
if (numbers[0].equals("6")) return "Discover Card";
else
if (numbers[0].equals("3") && numbers[1].equals("7")) return "American Express Card";

return "Invalid";
}

















}

File: python/credit_card_validator.py
def splitIntoArrays(numbers):

   return list(numbers)
  
  



def check_card_validity_status(array_numbers):
  new_array = []
  if len(array_numbers) < 13 or len(array_numbers) > 16:
      return "Invalid"
  
  for numbers in array_numbers[::2]:
    double_num = int(numbers) * 2
    
    if double_num >= 10:
      first_num = double_num % 10
      divide_one = double_num // 10
      second_num = divide_one % 10
      sum = first_num + second_num  
      new_array.append(sum)
    else:
       new_array.append(double_num)
  sum = 0
  for numbers in new_array:
   sum += numbers
  odd_sum = 0
  for elements in array_numbers[1::2]:
    odd_sum += int(elements)
  total_sum = odd_sum + sum 
  if total_sum % 10 == 0:
    return "Valid"

  return "Invalid"
      




def check_card_type(list_numbers):

   if list_numbers[0] == "5": 
        return "Master Card" 
   elif list_numbers[0] == "4": 
        return "Visa Card" 
   elif list_numbers[0] == "6": 
        return "Discover Card" 
   elif list_numbers[0] == "3" and list_numbers[1] == "7":
        return "American Express Card"
   else:
       return "Invalid"
         






def main_menu():

    user_input = input("Hello, Kindly Enter card details to verify \n")
    new_list = splitIntoArrays(user_input) 
    print("**Credit Card Type: ",check_card_type(new_list))
    print("**Credit Card Number: ",user_input)
    print("**Credit Digit length: ",len(new_list)) 
    print("**Credit Card Validity Status: ", check_card_validity_status(new_list ))
    print("**************************************")
main_menu()



















